packages/dara-components/js/graphs/graph-layout.tsx

```diff
@@ -258,6 +231,21 @@ export function parseLayoutDefinition(definition: GraphLayoutDefinition): GraphL
         }
     }
 
+    if (isDefinitionWithTiers(definition)) {
+        const builderWithTiers = builder as unknown as BuilderWithTiers;
```

Comment:
You could force TypeScript to treat `builder` as `BuilderWithTiers` by building another typeguard, so you'd do `isDefinitionWithTiers(definition) && isTiersBuilder(builder)`, if the BuilderWithTiers was recognizable from its shape (if not can attach a 'supportsTiers' property or similar)